students/luftsmeerflier/lesson05/mailroom_3.py

```diff
@@ -0,0 +1,123 @@
+#/usr/bin/env python3
+import sys
+
+donors = {
+	"John Quigley": {"2019-02-04": 5000.00, "2019-02-05": 4250.00, "2019-02-07": 3000.00},
+	"Sara Smith": {"2020-01-02": 2000.00, "2019-02-08": 42000.00, "2019-02-02": 32000.00},
+	"Jacob van der Schmidt": {"2012-01-20": 20.00, "2019-02-24": 2000.00, "2019-12-1": 3400.00},
+	"Ogden Nash": {"2019-06-04": 5000000.00, "2019-04-04": 500000.00, "2019-12-04": 50000.00}
+}
+
+def send_thank_you(first_name='', last_name=''):
+	names_list = list(donors.keys())
+
+	_list = input("Type `list` for a list of donors or press the carriage key to skip\n")
+
+	if _list == "list":
+		for donor in names_list:
+			print(donor)
+		print('\n')
+	else:
+		pass
+
+	name = input('What is the name of the donor? (First and Last i.e. "John Quigley"\n').title()
+	donor = donors.get(name)
+
+	if donor:
+		amount = list(donor.values())[0]
+		date = list(donor.keys())[0]
+		write_letters(name, amount, date)
+	else:
+		
+		amount = input("How much did {} donate?\n".format(name))
+		date = input("When was the donation? YYYY-MM-DD\n")
+		try:
+			if amount > 0:
+				print("That is an invalid donation amount")
+				return 1
+			else:
+				write_letters(name, amount, date)
+		except TypeError:
```

Comment:
this is always going to fail here - I am currently unable to enter a donation.

my suggestion is you convert your amount to float with a catch for ValueError.

```
try:
    amount = float(amount)
except ValueError:
   ...
```